apple-codesign/src/code_directory.rs

```diff
@@ -540,23 +540,8 @@ impl<'a> CodeDirectoryBlob<'a> {
     pub fn adjust_version(&mut self) -> u32 {
         let old_version = self.version;
 
-        let mut minimum_version = CodeDirectoryVersion::Initial;
+        let mut minimum_version = CodeDirectoryVersion::SupportsExecutableSegment;
```

Comment:
I'm not strongly opposed to this change. But could you please provide some justification for it? Ideally as an inline comment.

The reason the code is the way it is is to provide maximum flexibility and allow writing out all versions of code signatures using this crate. This change as is prevents doing that.

Reply:
so the justification is that ios 15+ requires this as the minimum version. not sure if we should make the minimum version configurable instead. Is there a use case for creating old versions? Do you know how old macosx or ios would have to be to not support this version?

Reply:
Thinking about this more and looking at changes Apple has made for newer OS versions, I think we need to build out better support for targeting a specific code directory version when signing. This likely entails teaching `SigningSettings` to hold a minimum version and then adding code to set reasonable defaults for fields on newer versions.

This is beyond the scope of this PR and is something I'll tackle myself. Hopefully I push some commits in the next few days.

Reply:
I implemented functionality for automatically adjusting the minimum code signature version in commit 83eb276d29d6233533a7730ebb8e52504058b461. As the commit message states, this is probably not comprehensive. But it should hopefully address iOS 15+ issues.